### DISCUSSION TITLE
How can I use Dynamic Segments and Splat Routes together?

### DISCUSSION THREAD

**Discussion by uttk** [remix-run/remix#9224] [Q&A] [answered]
There are two paths in routes

- `api.$.ts` → return json
- `$username.posts.tsx` → returns the HTML of `/$username/posts`.

At this point, the expected behaviour is as follows

- `api.$.ts` → matches `/api/*`
- `$username.posts.tsx`  → matches `/$username/posts` except `/api/*`

With that in mind, the actual behaviour when requesting `/api/posts` is as follows.

- `api.$.ts` → no matches
- `$username.posts.tsx` → matches and returns HTML

What can I do to get the expected behaviour?

---

**Versions**

```json
 "dependencies": {
    "@remix-run/node": "^2.8.1",
    "@remix-run/react": "^2.8.1",
    "@remix-run/serve": "^2.8.1",
  },
  "devDependencies": {
    "@remix-run/dev": "^2.8.1",
  },
```

**Comment by kiliman** [comment, marked as answer]
You're right. It appears to be a bug in the route matching. My understanding is that static segments like `api` have higher precedence over a path param.

So the URL `/api/posts` should always prefer `/api/*` over `/:username/posts/`

Here's a repro:

https://stackblitz.com/edit/remix-run-remix-h3nrmh?file=app%2Froutes%2F%24username.posts.tsx

**Reply by uttk** [reply]
Thank you kindly for your kindness!
I will throw the issue and see what happens.
